src/HazelcastClient.ts

```diff
@@ -469,7 +469,11 @@ export class HazelcastClient {
                 return this;
             })
             .catch((e) => {
-                logger.error('HazelcastClient', 'Client failed to start', e);
+                this.shutdown()
```

Comment:
Should we wait for this shutdown call before throwing immediately?

Reply:
There are two reasons why I implemented the shutdown-on-start-error in "async" way. First, `shutdown` call itself might take some time to finish. Second, `shutdown` may throw (reject to be more precise) and shadow the original error. It's possible to mitigate the at least the second problem, but I'm not sure if it's worth it. Do you see any benefit for the end user in "sync" approach?

Reply:
Well, at first I thought the shutdown call should be a quick one. I thought the only thing that could take some time is the shutdown call of the connection manager. If it is possible for this to take some time, then it makes sense to keep things as it is.

Reply:
Yes, `CPSubsystem` may take some time to release sessions.

Reply:
But, there should not be any sessions created during the startup

Reply:
Yes, I forgot to mention that the session clean up scenario is not currently possible, but that may change in future versions, as `shutdown` is async since 4.0.